### ISSUE TITLE
Full path in "entryname"

### ISSUE THREAD

**Issue by tetron** [common-workflow-language/cwl-v1.2#37]
I believe it is a de facto behavior of `cwltool` that if `entryname` under InitialWorkDir is a full path and it is running in a Docker container, can mount inputs anywhere on the filesystem that you need to.

I think we discussed this but I can't find any documentation or conformance tests.

This comes up fairly frequently ("how can I control mount points in Docker?") so we should probably formalize it.

**Comment by tetron** [comment]
An example is

```
inputs:
  inputsdir: Directory
requirements:
  DockerRequirement: { dockerPull: image }
  InitialWorkDirRequirement:
    listing:
      - entryname: /inputs
        entry: $(inputs.inputsdir)
```

Will mount `inputsdir` at `/inputs` instead of wherever the runner feels like putting it.

**Comment by kaushik-work** [comment]
@tetron I will check what SB executor does, and if there are any security/compliance issues with this. Thanks!

**Comment by bogdang989** [comment]
I would argue that this should only be acceptable if full path is under workdir (I think glob works like that too, full path is allowed but only pointing to path in the workdir). I am thinking various custom paths could be used in various systems for various things, so not sure it makes sense to allow this sort of behavior. Other than that, the InitialWorkDirRequirement should be applied to the workdir as the name says :)

**Comment by mr-c** [comment]
This trick is already out in the wild https://github.com/mr-c/biobox-cwl-example/blob/main/biobox-megahit.cwl

**Comment by tetron** [comment]
From discussion 2020-07-14: allow this.  add tests.  executors can disallow this if they want.

**Comment by tetron** [comment]
to be valid:

must have `DockerRequirement`

must be in `requirements`

will add positive/negative tests

**Comment by tetron** [comment]
PR under #39

**Comment by mr-c** [comment]
Fixed by #39
